autotest/include/io.h

```diff
@@ -14,6 +14,8 @@
 
 #if defined(__arm__)
 #include <arch/asm-cortex-m/io.h>
+#elif defined(CONFIG_ARCH_RV32)
+#include <arch/asm-rv32/io.h>
```

Comment:
asm-generic io.h (that hold the ifdef) to add, but can be made in the next PR :-) (enhancement). @lb-anssi if you're ok with that, can you just push an issue about the removal of arch-specific headers inclusion out of arch ?
If we merge your PR today I'll try to fix that tomorrow